vochain/state/process.go

```diff
@@ -211,9 +260,28 @@ func (v *State) UpdateProcess(p *models.Process, pid []byte) error {
 	if p == nil || len(p.ProcessId) != types.ProcessIDsize {
 		return ErrProcessNotFound
 	}
+	// update the process
 	v.Tx.Lock()
-	defer v.Tx.Unlock()
-	return updateProcess(&v.Tx, p, pid)
+	if err := updateProcess(&v.Tx, p, pid); err != nil {
+		return err
+	}
+	v.Tx.Unlock()
```

Comment:
if `err != nil` happens, Unlock will never happen. add it before the `return err` , or use defer, if the `try to update startBlocks database` doesn't take long

Reply:
or even simpler
```go
 	v.Tx.Lock()
	err := updateProcess(&v.Tx, p, pid)
    v.Tx.Unlock()
	if err != nil {
		return err
	}
```